targets/TARGET_Cypress/TARGET_PSOC6/sleep_api.c

```diff
@@ -26,16 +29,34 @@ void hal_sleep(void)
     Cy_SysPm_Sleep(CY_SYSPM_WAIT_FOR_INTERRUPT);
 }
 
+
+#define DEEP_SLEEP_ENTER_TIMEOUT_US     260000
```

Comment:
This potentially violates this requirement:
```
 *   * the wake-up time should be less than 10 ms - Verified by deepsleep_lpticker_test().
```

Reply:
My understanding is that this requirement assumes the hardware is in a state ready to enter deep sleep mode when the hal_deepsleep() function is called. But this seems not always to be satisfied and my code tries to implement a workaround. Ideally, HAL API should be able to return an error code when it is not able to enter deep sleep mode as requested.

Reply:
If not able to enter deepsleep, it should return immediately